### PR TITLE
Add Keypoints type to annotations and predictions

### PR DESCRIPTION
**Pull request by flubstep**
Keypoints labels are collections of points similar to polygon or line labels. They are firstly defined as a list of points, but each point is intended to match up to a static nomenclature (e.g. "chin", "left_ear", "right_ear") and also include a description of the intended display of this label via a collection of line segments called a "skeleton". An example of a keypoints label are facial keypoints, which can be seen on this example of a synthetic face:

image

### CODE REVIEW

**Comment by flubstep** [comment]
CC @drakejwong, still waiting on a server-side merge before the tests will run properly, but running this by early in order to make sure the code checks out.

File: nucleus/__init__.py
Comment (drakejwong) [inline nucleus/__init__.py:54]: we'll need to add this to `__all__` for it to start showing up in the docs ([preview here](https://scale-ai-nucleus-python-sdk--278.com.readthedocs.build/en/278/api/nucleus/index.html))

File: nucleus/__init__.py
Comment (drakejwong) [inline nucleus/__init__.py:127]: same here, we'll need to add this to `__all__` for it to start showing up in the docs ([preview here](https://scale-ai-nucleus-python-sdk--278.com.readthedocs.build/en/278/api/nucleus/index.html))

File: nucleus/annotation.py
Comment (drakejwong) [inline nucleus/annotation.py:467]: nit
```suggestion
        self.metadata = self.metadata or {}
```

File: nucleus/annotation.py
Comment (drakejwong) [inline nucleus/annotation.py:429]: not super familiar with typical OOP paradigms for keypoints, but filed a long term follow up to support string based skeletons like `[["point1", "point2"], ...]`

File: nucleus/annotation.py
Comment (drakejwong) [inline nucleus/annotation.py:428]: any reason not to embed this in the `Keypoint` object?

**Review by drakejwong — APPROVED** [review]
ty! left some small comments. as you mentioned we're also awaiting the backend merge but will stamp to unblock

File: nucleus/annotation.py
Comment (flubstep) [reply nucleus/annotation.py:428]: Yeah, I thought about that a little bit but ended up sticking with this solution. The reason why is that the labels themselves are really only the keypoints (the output of the network in COCO format is x, y, v), and so the names and skeleton are static across the entire project. Which means two things:

1. It's similar to how the taxonomy and the labels are separate in category annotations
2. It's easiest just to define your keypoint names + skeleton once and then just add them as a keyword in the constructor, as opposed to having to zip() the keypoint names against the x,y coordinates

Comment (drakejwong) [reply nucleus/annotation.py:428]: gotcha that makes sense, thanks!

**Pull request merged**
